pkg/querier/querier.go

```diff
@@ -128,25 +128,42 @@ func getChunksIteratorFunction(cfg Config) chunkIteratorFunc {
 	return mergeChunks
 }
 
+func queryIngesters(
```

Comment:
looking to expose these two funcs or something very similar after this for the work to split the request queues between queries that will use ingesters, ingesters + store gateways, and just store-gateways.

I also intend to these down in the queriers themselves, in `func (q *distributorQuerier) Select` and `func (q *blocksStoreQuerier) queryWithConsistencyCheck`, but the logic there is intertwined with clamping so I need to dig in deeper before applying them there.

So ideally the queue splitting, `NewQueryable`, and the queriers themselves would all use the exact same functions.

Reply:
I actually like your solution of moving these checks into `NewQueryable` call, it makes code easier to follow imho.